### PR TITLE
Issue #205: Stop storing IP of contributors in the logs

### PR DESCRIPTION
**Pull request by jpear1**
Someone will have to run `bin/cake migrations migrate` on the production server for this to take effect.

### CODE REVIEW

**Comment by jiru** [comment]
@jpear1 Good job with your first pull request :+1:

There are a few things that can be improved though.

1. You also need to update the `last_contributions` table. This table is holding a copy of the last entries of `contributions`. It has exactly the same structure, so just repeating the same schema update command should be enough. `last_contributions` is created and updated with the help of SQL code in `docs/database/triggers/last_contributions_triggers.sql`. This code needs to be updated accordingly.

2. You also need to update the test fixtures in `tests/Fixture/ContributionsFixture.php` (and `tests/Fixture/LastContributionsFixture.php`). I don’t know if you are familiar with [unit testing](https://en.wikipedia.org/wiki/Unit_testing), but we have a test suite located in the `tests` folder. The test suite is executed by PHPunit, which creates a dummy database (called `tatoeba_test`), fills it with dummy values, and use these values as a base to check if the code in `src` is working. These dummy values are what we call test fixtures, and we need to keep them synchronized with the actual database. So for example, the file `tests/TestCase/Model/Table/SentencesTableTest.php` is testing the code located in `src/Model/Table/SentencesTable.php`. You can run all or some specific tests like this:
    ```sh
    vagrant ssh
    cd Tatoeba/
    # run all the test suite
    phpunit
    # run a specific test file
    phpunit tests/TestCase/Model/Table/SentencesTableTest.php
    # run a specific test in a specific file
    phpunit tests/TestCase/Model/Table/SentencesTableTest.php --filter testLogsSentenceDeletionOnDelete
    ```
    We use these tests to make sure any change does not break existing functionality. The little green check mark next to your commit on Github shows that your modifications did not break anything so far.

3. You should remove all the code related to the storage of IPs, even though just removing the field is enough to get things working. For example, the `saveSentenceContribution()` method in `src/Model/Table/ContributionsTable.php` is still trying to store the IP address:
    https://github.com/Tatoeba/tatoeba2/blob/dc86c5e96a3047721d5b5ee3ccab25edec5357fd/src/Model/Table/ContributionsTable.php#L267-L283
    Similarly, `CurrentUser::getIp()` can be safely removed because it’s not used anywhere else.

**Comment by jpear1** [comment]
I found a usage of contributions' IP addresses being used and I'm not sure how to safely remove it:

https://github.com/Tatoeba/tatoeba2/blob/119c63774b3e9ee62eec77c02f8d7b442d8d35f6/src/Controller/Component/MailerComponent.php#L47-L68

It seems like `Email` requires IP addresses... but it's a CakePHP class and IP addresses aren't mentioned in the docs. Does `viewVars` actually create new members of the `Email` class that are used somewhere else in the program?

**Comment by AndiPersti** [comment]
> It seems like `Email` requires IP addresses... but it's a CakePHP class and IP addresses aren't mentioned in the docs. Does `viewVars` actually create new members of the `Email` class that are used somewhere else in the program? Maybe when generating the text of the email?

The `Email` class allows you to use a template for the email body (see [line 52](https://github.com/Tatoeba/tatoeba2/blob/119c63774b3e9ee62eec77c02f8d7b442d8d35f6/src/Controller/Component/MailerComponent.php#L52)). And in order to pass variables to that template you can use the `viewVars` method. So all the members in the array are variable names that are available in the template `src/Template/Email/html/blocked_or_suspended_user.ctp`.

I don't think having the IPs in the email is necessary because the recipients are community admins that don't have access to the server in order to e.g. block these IPs. And a sysadmin can get the IPs through other means.

Thus, I would simply remove the IPs from the email message.

**Comment by jiru** [comment]
@jpear1 Good catch! Yes, you can just remove the part that lists IPs from the email template.

**Comment by jiru** [comment]
@jpear1 I see that you removed the IPs from the list of variables passed to the `blocked_or_suspended_user` template. That all good, but you also need to modify that template to remove the part that mentions the IP addresses. Otherwise, that will produce "undefined variable" warnings.

File: src/Model/Table/ContributionsTable.php [hunk 1]
```diff
@@ -321,9 +318,8 @@ public function excludeBots($query)
     public function getLastContributionOf($userId)
     {
         return $this->find()
-            ->select(['ip', 'count' => 'count(*)'])
+            ->select(['count' => 'count(*)'])
             ->where(['user_id' => $userId])
-            ->group(['group' => 'ip'])
             ->orderDesc('count')
             ->limit(10)
             ->all();
```
Comment (jiru) [inline src/Model/Table/ContributionsTable.php:325]: This method is not used anywhere else, so you could as well remove it altogether.

Comment (jpear1) [reply src/Model/Table/ContributionsTable.php]: Alright, done.

**Comment by jpear1** [comment]
Added suggested changes.

**Comment by jiru** [comment]
Thank you, it’s all good! :+1:

**Pull request merged**